Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/SettingsFile.java

```diff
@@ -336,6 +336,18 @@ else if ((current != null))
 			}
 		}
 
+		if (fileName.equals(SettingsFile.FILE_NAME_DOLPHIN) && sections.isEmpty())
```

Comment:
Rather than only adding the GCPad keys if there are no sections at all in the file, I think you should add the GCPad keys as long as the GCPad keys are missing.

Reply:
@JosJuice you are right, fixed it :)